src/Esfa.Vacancy.Register.Application/Queries/GetVacancy/GetVacancyQueryHandler.cs

```diff
@@ -35,6 +40,21 @@ public class GetVacancyQueryHandler : IAsyncRequestHandler<GetVacancyRequest, Ge
 
             if (vacancy == null) throw new ResourceNotFoundException($"Vacancy: {message.Reference}");
 
+            if (vacancy.FrameworkCode.HasValue)
+            {
+                var framework = await _trainingDetailService.GetFrameworkDetails(vacancy.FrameworkCode.Value);
```

Comment:
I will not be able to use the client interfaces as we need to provide the base URLs which is only accepted via constructor of the implementation class.